### PR TITLE
tracee: add event derivation registering

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Copy and paste the git log here.

Fixes: #issue_number

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [x] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: pkg/events/derive/derive.go [hunk 1]
```diff
@@ -5,26 +5,43 @@ import (
...
+
+// Register registers a new derivation handler
+func (t Table) Register(deriveFrom, deriveTo events.ID, deriveCondition func() bool, deriveLogic DeriveFunction) {
+	if t[deriveFrom] == nil {
+		t[deriveFrom] = make(map[events.ID]struct {
+			DeriveFunction DeriveFunction
+			Enabled        func() bool
+		})
+	}
+	t[deriveFrom][deriveTo] = struct {
```
Comment (yanivagman) [inline pkg/events/derive/derive.go:29]: Maybe we should handle the case that (deriveFrom, DeriveTo) pair is already registered and return an error in that case

Comment (NDStrahilevitz) [reply pkg/events/derive/derive.go:29]: I thought I missed some validation here, thanks for catching it.

**Comment by rafaeldtinoco** [comment]
@NDStrahilevitz needs rebase and I'd like to review it tomorrow (and double-check with Geyslan as he has higher priority for merges currently).

**Comment by geyslan** [comment]
Despite this one requiring rebase, I was checking it against multi-scopes and noticed that it touches a file removed by https://github.com/aquasecurity/tracee/commit/168ec049d9d7ff17982c983741f818e94a1fb081.

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
Lets wait for multi scopes, rebase this and solve conflicts then.

**Comment by yanivagman** [comment]
This commit is already in main. A simple rebase will suffice

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**